samples/dnn/text_detection.py

```diff
@@ -60,179 +68,88 @@ def fourPointsTransform(frame, vertices):
-    cap = cv.VideoCapture(args.input if args.input else 0)
-
-    tickmeter = cv.TickMeter()
-    while cv.waitKey(1) < 0:
-        # Read frame
-        hasFrame, frame = cap.read()
-        if not hasFrame:
-            cv.waitKey()
-            break
-
-        # Get frame height and width
-        height_ = frame.shape[0]
-        width_ = frame.shape[1]
-        rW = width_ / float(inpWidth)
-        rH = height_ / float(inpHeight)
-
-        # Create a 4D blob from frame.
-        blob = cv.dnn.blobFromImage(frame, 1.0, (inpWidth, inpHeight), (123.68, 116.78, 103.94), True, False)
-
-        # Run the detection model
-        detector.setInput(blob)
-
-        tickmeter.start()
-        outs = detector.forward(outNames)
-        tickmeter.stop()
-
-        # Get scores and geometry
-        scores = outs[0]
-        geometry = outs[1]
-        [boxes, confidences] = decodeBoundingBoxes(scores, geometry, confThreshold)
-
-        # Apply NMS
-        indices = cv.dnn.NMSBoxesRotated(boxes, confidences, confThreshold, nmsThreshold)
-        for i in indices:
-            # get 4 corners of the rotated rect
-            vertices = cv.boxPoints(boxes[i])
-            # scale the bounding box coordinates based on the respective ratios
-            for j in range(4):
-                vertices[j][0] *= rW
-                vertices[j][1] *= rH
-
-
-            # get cropped image using perspective transform
-            if modelRecognition:
-                cropped = fourPointsTransform(frame, vertices)
-                cropped = cv.cvtColor(cropped, cv.COLOR_BGR2GRAY)
-
-                # Create a 4D blob from cropped image
-                blob = cv.dnn.blobFromImage(cropped, size=(100, 32), mean=127.5, scalefactor=1 / 127.5)
-                recognizer.setInput(blob)
-
-                # Run the recognition model
-                tickmeter.start()
-                result = recognizer.forward()
-                tickmeter.stop()
-
-                # decode the result into text
-                wordRecognized = decodeText(result)
-                cv.putText(frame, wordRecognized, (int(vertices[1][0]), int(vertices[1][1])), cv.FONT_HERSHEY_SIMPLEX,
-                           0.5, (255, 0, 0))
-
-            for j in range(4):
-                p1 = (int(vertices[j][0]), int(vertices[j][1]))
-                p2 = (int(vertices[(j + 1) % 4][0]), int(vertices[(j + 1) % 4][1]))
-                cv.line(frame, p1, p2, (0, 255, 0), 1)
-
-        # Put efficiency information
-        label = 'Inference time: %.2f ms' % (tickmeter.getTimeMilli())
-        cv.putText(frame, label, (0, 15), cv.FONT_HERSHEY_SIMPLEX, 0.5, (0, 255, 0))
-
-        # Display the frame
-        cv.imshow(kWinName, frame)
-        tickmeter.reset()
+    args = get_args_parser()
+    args.model = findFile(args.model)
+    args.input = findFile(args.input)
+    args.vocabulary_path = findFile(args.vocabulary_path)
+
+    frame = cv2.imread(args.input)
+
+    if(args.alias == "DB"):
```

Comment:
There is no alias option in command line params

Reply:
@asmorkalov The alias option is added using the models.yml via args.zoo

```python
    args, _ = parser.parse_known_args()
    add_preproc_args(args.zoo, parser, 'text_detection')
```

Reply:
We need to provide the detection model name and path to recognition model. I have updated the usage made the path to recognition model as required

`python text_detection.py DB --ocr=<path to recognition model>`